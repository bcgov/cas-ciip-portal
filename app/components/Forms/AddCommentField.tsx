import React, {useState} from 'react';
import {FieldProps} from '@rjsf/core';
import {Form, Button} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faCommentAlt} from '@fortawesome/free-solid-svg-icons';
import {toast} from 'react-toastify';

const AddCommentField: React.FunctionComponent<FieldProps<string>> = (
  props
) => {
  const [expanded, setExpanded] = useState(false);
  const [value, setValue] = useState(props.formData);
  const title = props.formData ? 'Edit comment' : 'Add a comment';

  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setValue(e.target.value);
  };

  const handleCancel = () => {
    setValue(props.formData);
    setExpanded(false);
  };

  const handleSave = () => {
    props.onChange(value);
    setExpanded(false);
    toast.success(
      'Your comment was saved for review upon application submission',
      {
        autoClose: 5000,
        position: 'top-center'
      }
    );
  };

  return (
    <>
      <style jsx>{`
        .title,
        .report-field .controls {
          display: flex;
          width: 100%;
          justify-content: flex-end;
          align-items: center;
        }

        .report-field .controls {
          margin-top: 16px;
        }

        .report-field {
          background-color: #fcf8e2;
          border: 1px solid #ccc;
          border-radius: 4px;
          padding: 16px;
          margin-top: 16px;
        }
      `}</style>
      <span className="title">
        <Button variant="link" onClick={() => setExpanded(true)}>
          <FontAwesomeIcon icon={faCommentAlt} />
          &nbsp;
          {title}
        </Button>
      </span>
      {expanded && (
        <div className="report-field">
          <legend>CIIP Application Comment</legend>

          <p>
            This area is only for leaving a comment regarding the data that you
            reported on this page. This comment will be visible by the person
            reviewing this CIIP application once the application is submitted.
          </p>
          <p>
            For any technical issue, or to review documentation, please click on
            the help bubble in the bottom right of your screen.
          </p>
          <Form.Control
            as="textarea"
            rows={8}
            value={value}
            onChange={handleChange}
          />
          <div className="controls">
            <Button variant="outline-dark" onClick={handleCancel}>
              Cancel
            </Button>
            <Button
              style={{marginLeft: 16}}
              variant="success"
              onClick={handleSave}
            >
              Save Comment
            </Button>
          </div>
        </div>
      )}
    </>
  );
};

export default AddCommentField;
