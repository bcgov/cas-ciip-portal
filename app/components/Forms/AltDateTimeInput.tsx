// https://github.com/rjsf-team/react-jsonschema-form/blob/v1.8.1/src/components/widgets/AltDateTimeWidget.js

import React from "react";
import PropTypes from "prop-types";
import AltDateInput from "./AltDateInput";
import { WidgetProps } from "react-jsonschema-form";

function AltDateTimeInput(props: WidgetProps) {
  const { AltDateInput } = props.registry.widgets;
  return <AltDateInput time {...props} />;
}

if (process.env.NODE_ENV !== "production") {
  AltDateTimeInput.propTypes = {
    schema: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    value: PropTypes.string,
    required: PropTypes.bool,
    onChange: PropTypes.func,
    options: PropTypes.object,
  };
}

AltDateTimeInput.defaultProps = {
  ...AltDateInput.defaultProps,
  time: true,
};

export default AltDateTimeInput;