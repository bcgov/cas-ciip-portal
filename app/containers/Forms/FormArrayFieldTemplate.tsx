import React from 'react';
import {ArrayFieldTemplateProps} from 'react-jsonschema-form';
import {Form, Button, Col} from 'react-bootstrap';

const FormArrayFieldTemplate: React.FunctionComponent<ArrayFieldTemplateProps> = ({
  items,
  canAdd,
  onAddClick,
  uiSchema,
  className
}) => {
  return (
    <div className={className}>
      {items.map(element => {
        return (
          <React.Fragment key={element.index}>
            <Form.Row>
              <Col xs={12} md={element.hasRemove ? 10 : 12}>
                {element.children}
              </Col>

              {element.hasRemove && (
                <Col xs={12} md={2}>
                  <div className="remove-button-container">
                    <Button
                      type="button"
                      variant="danger"
                      onClick={element.onDropIndexClick(element.index)}
                    >
                      {uiSchema['ui:remove-text'] || 'Remove'}
                    </Button>
                  </div>
                </Col>
              )}
            </Form.Row>
            <hr />
          </React.Fragment>
        );
      })}
      {canAdd && (
        <Form.Row>
          <Col xs={12}>
            <div className="add-button-container">
              <Button type="button" onClick={onAddClick}>
                {uiSchema['ui:add-text'] || 'Add'}
              </Button>
            </div>
          </Col>
        </Form.Row>
      )}
      <style jsx>
        {`
          .remove-button-container {
            height: 100%;
            display: flex;
            justify-content: space-around;
            align-items: center;
          }
          .add-button-container {
            height: 100%;
            display: flex;
            justify-content: space-around;
            align-items: center;
          }
        `}
      </style>
    </div>
  );
};

export default FormArrayFieldTemplate;
