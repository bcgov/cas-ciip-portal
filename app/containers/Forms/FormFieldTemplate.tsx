import React from 'react';
import {Col, Form} from 'react-bootstrap';
import {FieldTemplateProps} from 'react-jsonschema-form';

const FormFieldTemplate: React.FunctionComponent<FieldTemplateProps> = ({
  id,
  label,
  help,
  required,
  description,
  errors,
  children,
  schema,
  classNames,
  uiSchema,
  hidden
}) => {
  if (hidden) return <div className="hidden">{children}</div>;

  const helpText = uiSchema['ui:helplink'] ? (
    <a
      className="font-italic"
      href={uiSchema['ui:helplink']}
      target="_blank"
      rel="noopener noreferrer"
    >
      {help}
    </a>
  ) : (
    <span className="font-italic text-muted">{help}</span>
  );

  if (schema.type === 'array')
    return (
      <Col xs={12}>
        {description}
        {helpText}
        {errors}
        {children}
      </Col>
    );
  if (schema.type === 'object')
    return (
      <Col xs={12}>
        <Form.Row>
          {description}
          {errors}
          {helpText}
          {children}
        </Form.Row>
      </Col>
    );

  const xs = uiSchema['ui:col-xs'] || 12;
  const sm = uiSchema['ui:col-sm'] || xs;
  const md = uiSchema['ui:col-md'] || (sm < 6 ? sm : 6);
  const lg = uiSchema['ui:col-lg'] || md;
  const xl = uiSchema['ui:col-xl'] || lg;

  return (
    <Form.Group
      as={Col}
      xs={xs}
      sm={sm}
      md={md}
      lg={lg}
      xl={xl}
      className={classNames}
    >
      <Form.Label htmlFor={id}>
        {label}
        {required ? <>&nbsp;*</> : null}
      </Form.Label>
      {description}
      {children}
      {errors}
      {helpText}
    </Form.Group>
  );
};

export default FormFieldTemplate;
