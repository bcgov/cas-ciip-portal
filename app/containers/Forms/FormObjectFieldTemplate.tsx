import React from 'react';
import {Col} from 'react-bootstrap';

const FormObjectFieldTemplate = ({
  TitleField,
  properties,
  title,
  description
}) => {
  return (
    <>
      {title && (
        <Col xs={12}>
          <TitleField title={title} />
        </Col>
      )}
      {properties.map(prop => (
        <React.Fragment key={prop.content.key}>{prop.content}</React.Fragment>
      ))}
      {description}
    </>
  );
};

export default FormObjectFieldTemplate;
