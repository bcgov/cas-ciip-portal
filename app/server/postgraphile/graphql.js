const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
