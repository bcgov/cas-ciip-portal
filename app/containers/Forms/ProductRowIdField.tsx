import React, {useMemo} from 'react';
import {FieldProps} from 'react-jsonschema-form';
import {createFragmentContainer, graphql} from 'react-relay';
import {ProductRowIdField_query} from 'ProductRowIdField_query.graphql';

interface Props extends FieldProps<number> {
  query: ProductRowIdField_query;
}

/**
 * Displays the information related to a product, and loads the additional data schema for products that need it.
 * By using the registry (which contains, for instance, the form's FieldTemplate and StringField)
 * this allows for this component to be reused in both the Form and the ApplicationDetailsCardItem components.
 * It will either render inputs or the values based on what is defined in the current jsonschema form registry.
 */
export const ProductRowIdFieldComponent: React.FunctionComponent<Props> = props => {
  /**
   * Injects the list of products in the schema, and remove `query` from the props
   * Other props are passed as-is to the StringField
   */
  const fieldProps = useMemo(
    () => ({
      ...props,
      schema: {
        ...props.schema,
        enum: props.query.allProducts.edges.map(({node}) => node.rowId),
        enumNames: props.query.allProducts.edges.map(({node}) => node.name),
        state: props.query.allProducts.edges.map(({node}) => node.state)
      },
      query: undefined
    }),
    [props]
  );

  return <props.registry.fields.StringField {...fieldProps} />;
};

export default createFragmentContainer(ProductRowIdFieldComponent, {
  query: graphql`
    fragment ProductRowIdField_query on Query {
      allProducts(condition: {state: "active"}) {
        edges {
          node {
            rowId
            name
            state
          }
        }
      }
    }
  `
});
