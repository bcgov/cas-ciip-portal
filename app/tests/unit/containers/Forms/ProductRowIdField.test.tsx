import React from 'react';
import {shallow} from 'enzyme';
import {ProductRowIdFieldComponent} from 'containers/Forms/ProductRowIdField';
import productionSchema from 'schema/data/prod/form_json/production.json';
import {JSONSchema7} from 'json-schema';
import {getDefaultRegistry} from '@rjsf/core/dist/cjs/utils';
import FormArrayFieldTemplate from 'containers/Forms/FormArrayFieldTemplate';
import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
import {ProductRowIdField_query} from '__generated__/ProductRowIdField_query.graphql';
import {ProductRowIdField_naicsCode} from '__generated__/ProductRowIdField_naicsCode.graphql';

describe('The ProductionRowIdField Component with published product matched to a naics code', () => {
  const query: ProductRowIdField_query = {
    ' $refType': 'ProductRowIdField_query',
    published: {
      edges: [
        {
          node: {
            rowId: 1,
            productName: 'foo',
            productState: 'PUBLISHED',
            isEnergyProduct: false
          }
        },
        {
          node: {
            rowId: 2,
            productName: 'dontrenderme',
            productState: 'PUBLISHED',
            isEnergyProduct: false
          }
        }
      ]
    },
    archived: {
      edges: [
        {
          node: {
            rowId: 3,
            productName: 'bar',
            productState: 'ARCHIVED'
          }
        }
      ]
    }
  };
  const naicsCode: ProductRowIdField_naicsCode = {
    ' $refType': 'ProductRowIdField_naicsCode',
    id: 'abc',
    productsByNaicsCode: {
      edges: [
        {
          node: {
            rowId: 1,
            productName: 'foo',
            productState: 'PUBLISHED'
          }
        }
      ]
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode,
    formData: 1,
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain the products in productsByNaicsCode when passed a naicsCode', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper.find('StringField').prop('schema' as any).enumNames[0]).toBe(
      'foo'
    );
  });

  it('should not contain the products not in productsByNaicsCode when passed a naicsCode', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper.find('StringField').prop('schema' as any).enumNames[1]).toBe(
      undefined
    );
  });
});

describe('The ProductionRowIdField Component with archived product', () => {
  const query: ProductRowIdField_query = {
    ' $refType': 'ProductRowIdField_query',
    published: {
      edges: []
    },
    archived: {
      edges: [
        {
          node: {
            rowId: 1,
            productName: 'bar',
            productState: 'ARCHIVED'
          }
        }
      ]
    }
  };
  const naicsCode: ProductRowIdField_naicsCode = {
    ' $refType': 'ProductRowIdField_naicsCode',
    id: 'abc',
    productsByNaicsCode: {
      edges: []
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode,
    formData: 1,
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the product name in an input group', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper.find('InputGroup').text()).toBe('bar');
  });
});

describe('The ProductionFields Component without naics code', () => {
  const query: ProductRowIdField_query = {
    ' $refType': 'ProductRowIdField_query',
    published: {
      edges: [
        {
          node: {
            rowId: 1,
            productName: 'foo',
            productState: 'PUBLISHED',
            isEnergyProduct: false
          }
        },
        {
          node: {
            rowId: 1,
            productName: 'dontrenderme',
            productState: 'PUBLISHED',
            isEnergyProduct: false
          }
        }
      ]
    },
    archived: {
      edges: [
        {
          node: {
            rowId: 2,
            productName: 'bar',
            productState: 'ARCHIVED'
          }
        }
      ]
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode: null,
    formData: 1,
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the product name in an input group', () => {
    const wrapper = shallow(<ProductRowIdFieldComponent {...props} />);
    expect(wrapper.find('InputGroup').text()).toBe('foo');
  });
});
