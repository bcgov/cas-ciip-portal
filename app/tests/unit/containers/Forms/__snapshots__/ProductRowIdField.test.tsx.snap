// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ProductionFields Component when rendered from the LinkProduct modal should match the snapshot 1`] = `
<StringField
  autofocus={false}
  disabled={false}
  errorSchema={null}
  formContext={Object {}}
  formData={1}
  idSchema={
    Object {
      "$id": "product_0",
    }
  }
  isLinkModal={true}
  naicsCode={null}
  name="product"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  readonly={false}
  registry={
    Object {
      "ArrayFieldTemplate": [Function],
      "FieldTemplate": [Function],
      "ObjectFieldTemplate": [Function],
      "definitions": Object {},
      "fields": Object {
        "AnyOfField": [Function],
        "ArrayField": [Function],
        "BooleanField": [Function],
        "DescriptionField": [Function],
        "NullField": [Function],
        "NumberField": [Function],
        "ObjectField": [Function],
        "OneOfField": [Function],
        "SchemaField": [Function],
        "StringField": [Function],
        "TitleField": [Function],
        "UnsupportedField": [Function],
      },
      "formContext": Object {},
      "rootSchema": Object {},
      "widgets": Object {
        "AltDateTimeWidget": [Function],
        "AltDateWidget": [Function],
        "BaseInput": [Function],
        "CheckboxWidget": [Function],
        "CheckboxesWidget": [Function],
        "ColorWidget": [Function],
        "DateTimeWidget": [Function],
        "DateWidget": [Function],
        "EmailWidget": [Function],
        "FileWidget": [Function],
        "HiddenWidget": [Function],
        "PasswordWidget": [Function],
        "RadioWidget": [Function],
        "RangeWidget": [Function],
        "SelectWidget": [Function],
        "TextWidget": [Function],
        "TextareaWidget": [Function],
        "URLWidget": [Function],
        "UpDownWidget": [Function],
      },
    }
  }
  required={false}
  schema={
    Object {
      "dependencies": Object {
        "requiresEmissionAllocation": Object {
          "oneOf": Array [
            Object {
              "properties": Object {
                "requiresEmissionAllocation": Object {
                  "const": false,
                },
              },
            },
            Object {
              "properties": Object {
                "productEmissions": Object {
                  "minimum": 0,
                  "title": "Emissions allocated to product",
                  "type": "number",
                },
                "requiresEmissionAllocation": Object {
                  "const": true,
                },
              },
              "required": Array [
                "productEmissions",
              ],
            },
          ],
        },
        "requiresProductAmount": Object {
          "oneOf": Array [
            Object {
              "properties": Object {
                "productAmount": Object {
                  "minimum": 0,
                  "title": "Annual Production Amount",
                  "type": "number",
                },
                "productUnits": Object {
                  "title": "Units",
                  "type": "string",
                },
                "requiresProductAmount": Object {
                  "const": true,
                },
              },
              "required": Array [
                "productAmount",
              ],
            },
            Object {
              "properties": Object {
                "requiresProductAmount": Object {
                  "const": false,
                },
              },
            },
          ],
        },
      },
      "enum": Array [
        1,
      ],
      "enumNames": Array [
        "foo",
      ],
      "properties": Object {
        "comments": Object {
          "type": "string",
        },
        "isEnergyProduct": Object {
          "type": "boolean",
        },
        "productRowId": Object {
          "title": "Product or Service",
          "type": "integer",
        },
        "requiresEmissionAllocation": Object {
          "type": "boolean",
        },
        "requiresProductAmount": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "productRowId",
        "requiresEmissionAllocation",
        "requiresProductAmount",
      ],
      "type": "object",
    }
  }
  uiSchema={
    Object {
      "items": Object {
        "comments": Object {
          "ui:col-md": 12,
          "ui:field": "ProblemReportField",
          "ui:options": Object {
            "disableRenderingIfEmpty": true,
            "label": false,
          },
        },
        "isEnergyProduct": Object {
          "ui:widget": "hidden",
        },
        "productAmount": Object {
          "ui:col-md": 4,
          "ui:help": "Report the annual production for the product, following the guidance document provided",
        },
        "productEmissions": Object {
          "ui:col-md": 4,
          "ui:help": "Report emissions associated to CIIP product, in tCO2e",
        },
        "productRowId": Object {
          "ui:col-md": 12,
          "ui:field": "productRowId",
          "ui:widget": "SearchWidget",
        },
        "productUnits": Object {
          "ui:col-md": 4,
          "ui:readonly": true,
        },
        "requiresEmissionAllocation": Object {
          "ui:widget": "hidden",
        },
        "requiresProductAmount": Object {
          "ui:widget": "hidden",
        },
        "ui:field": "product",
        "ui:order": Array [
          "comments",
          "productRowId",
          "productAmount",
          "productUnits",
          "*",
        ],
      },
      "ui:add-text": "Add a Product",
      "ui:remove-text": "Remove Product",
    }
  }
/>
`;

exports[`The ProductionFields Component without naics code and isLinkModal is false should match the snapshot 1`] = `
<Alert
  closeLabel="Close alert"
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "displayName": "Fade",
      "render": [Function],
    }
  }
  variant="danger"
>
  We did not find any products associated with your NAICS code. Please verify the NAICS code entered in the Admin tab exists and is correct.
</Alert>
`;

exports[`The ProductionRowIdField Component with archived product should match the snapshot 1`] = `
<div>
  <InputGroup
    className="mb-3"
  >
    <InputGroupPrepend>
      <InputGroupText>
        bar
      </InputGroupText>
    </InputGroupPrepend>
  </InputGroup>
</div>
`;

exports[`The ProductionRowIdField Component with published product matched to a naics code should match the snapshot 1`] = `
<StringField
  autofocus={false}
  disabled={false}
  errorSchema={null}
  formContext={Object {}}
  formData={1}
  idSchema={
    Object {
      "$id": "product_0",
    }
  }
  naicsCode={
    Object {
      " $refType": "ProductRowIdField_naicsCode",
      "id": "abc",
      "productsByNaicsCode": Object {
        "edges": Array [
          Object {
            "node": Object {
              "productName": "foo",
              "productState": "PUBLISHED",
              "rowId": 1,
            },
          },
        ],
      },
    }
  }
  name="product"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  readonly={false}
  registry={
    Object {
      "ArrayFieldTemplate": [Function],
      "FieldTemplate": [Function],
      "ObjectFieldTemplate": [Function],
      "definitions": Object {},
      "fields": Object {
        "AnyOfField": [Function],
        "ArrayField": [Function],
        "BooleanField": [Function],
        "DescriptionField": [Function],
        "NullField": [Function],
        "NumberField": [Function],
        "ObjectField": [Function],
        "OneOfField": [Function],
        "SchemaField": [Function],
        "StringField": [Function],
        "TitleField": [Function],
        "UnsupportedField": [Function],
      },
      "formContext": Object {},
      "rootSchema": Object {},
      "widgets": Object {
        "AltDateTimeWidget": [Function],
        "AltDateWidget": [Function],
        "BaseInput": [Function],
        "CheckboxWidget": [Function],
        "CheckboxesWidget": [Function],
        "ColorWidget": [Function],
        "DateTimeWidget": [Function],
        "DateWidget": [Function],
        "EmailWidget": [Function],
        "FileWidget": [Function],
        "HiddenWidget": [Function],
        "PasswordWidget": [Function],
        "RadioWidget": [Function],
        "RangeWidget": [Function],
        "SelectWidget": [Function],
        "TextWidget": [Function],
        "TextareaWidget": [Function],
        "URLWidget": [Function],
        "UpDownWidget": [Function],
      },
    }
  }
  required={false}
  schema={
    Object {
      "dependencies": Object {
        "requiresEmissionAllocation": Object {
          "oneOf": Array [
            Object {
              "properties": Object {
                "requiresEmissionAllocation": Object {
                  "const": false,
                },
              },
            },
            Object {
              "properties": Object {
                "productEmissions": Object {
                  "minimum": 0,
                  "title": "Emissions allocated to product",
                  "type": "number",
                },
                "requiresEmissionAllocation": Object {
                  "const": true,
                },
              },
              "required": Array [
                "productEmissions",
              ],
            },
          ],
        },
        "requiresProductAmount": Object {
          "oneOf": Array [
            Object {
              "properties": Object {
                "productAmount": Object {
                  "minimum": 0,
                  "title": "Annual Production Amount",
                  "type": "number",
                },
                "productUnits": Object {
                  "title": "Units",
                  "type": "string",
                },
                "requiresProductAmount": Object {
                  "const": true,
                },
              },
              "required": Array [
                "productAmount",
              ],
            },
            Object {
              "properties": Object {
                "requiresProductAmount": Object {
                  "const": false,
                },
              },
            },
          ],
        },
      },
      "enum": Array [
        1,
      ],
      "enumNames": Array [
        "foo",
      ],
      "properties": Object {
        "comments": Object {
          "type": "string",
        },
        "isEnergyProduct": Object {
          "type": "boolean",
        },
        "productRowId": Object {
          "title": "Product or Service",
          "type": "integer",
        },
        "requiresEmissionAllocation": Object {
          "type": "boolean",
        },
        "requiresProductAmount": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "productRowId",
        "requiresEmissionAllocation",
        "requiresProductAmount",
      ],
      "type": "object",
    }
  }
  uiSchema={
    Object {
      "items": Object {
        "comments": Object {
          "ui:col-md": 12,
          "ui:field": "ProblemReportField",
          "ui:options": Object {
            "disableRenderingIfEmpty": true,
            "label": false,
          },
        },
        "isEnergyProduct": Object {
          "ui:widget": "hidden",
        },
        "productAmount": Object {
          "ui:col-md": 4,
          "ui:help": "Report the annual production for the product, following the guidance document provided",
        },
        "productEmissions": Object {
          "ui:col-md": 4,
          "ui:help": "Report emissions associated to CIIP product, in tCO2e",
        },
        "productRowId": Object {
          "ui:col-md": 12,
          "ui:field": "productRowId",
          "ui:widget": "SearchWidget",
        },
        "productUnits": Object {
          "ui:col-md": 4,
          "ui:readonly": true,
        },
        "requiresEmissionAllocation": Object {
          "ui:widget": "hidden",
        },
        "requiresProductAmount": Object {
          "ui:widget": "hidden",
        },
        "ui:field": "product",
        "ui:order": Array [
          "comments",
          "productRowId",
          "productAmount",
          "productUnits",
          "*",
        ],
      },
      "ui:add-text": "Add a Product",
      "ui:remove-text": "Remove Product",
    }
  }
/>
`;
